### PR TITLE
Canvas to image improvements / Safari images fix

### LINKED ISSUE #8923: Some images missing for thumbnails in Safari
## Bug Description

Originally reported by @choumx 
In Safari it appears that some of the image thumbnails are missing:
image26

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by merapi**
## Summary

This fixes Safari missing images issue due to unfinished image decoding (the library was waiting an arbitrary 100ms to take the "screenshot", after an exhaustive investigation it turns out that adding more delay (and doing various other things), only partially fixes the issue [it happened less often]).
Thanks to that discovery we can also remove the 100ms delay, speeding up all `html-to-image` usages (Presets/Text Sets smart color + Eyedropper + Thumbnails caching) 🎉 

## To-do

- [ ] Upstream PRs

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

Can be also tested on other browsers but the Safari is the main goal.
1. Enable thumbnails caching in the Experiments
2. Create a story and insert a few pages with images
3. Check if thumbnails are generated properly (images not missing)

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8923

### CODE REVIEW

File: packages/story-editor/src/components/colorPicker/basicColorPicker.js [hunk 1]
```diff
@@ -97,7 +97,7 @@ function BasicColorPicker({
           allowsGradient={allowsGradient}
           aria-labelledby="colorpicker-default-colors-title"
         />
-        {allowsSavedColors && savedColors.length && (
+        {allowsSavedColors && savedColors.length > 0 && (
```
Comment (merapi) [inline packages/story-editor/src/components/colorPicker/basicColorPicker.js:100]: When I was resetting the site I saw this:
Screenshot 2021-09-27 at 16 30 45

File: packages/story-editor/src/elements/media/imageDisplay.js [hunk 2]
```diff
@@ -107,6 +107,7 @@ function ImageDisplay({ element, box, previewMode }) {
       <Img
         ref={ref}
         draggable={false}
+        decoding="sync"
```
Comment (merapi) [inline packages/story-editor/src/elements/media/imageDisplay.js:110]: Adding it also here shouldn't cause any issues.

File: patches/html-to-image+1.8.5.patch [hunk 3]
```diff
@@ -0,0 +1,41 @@
+diff --git a/node_modules/html-to-image/.DS_Store b/node_modules/html-to-image/.DS_Store
```
Comment (swissspidy) [inline patches/html-to-image+1.8.5.patch:1]: Is there an upstream PR for this patch so that we don't have to maintain it forever?

**Review by miina — APPROVED** [review]
Awesome work!

The thumbnails work as expected in Safar as far as I tested.

(Logged an unrelated issue while testing: #9173)

**Comment by miina** [comment]
@merapi Some tests seem to be failing (not sure if related?) and there are conflicts to resolve now, too.

**Review by barklund — APPROVED** [review]
Wow, that's an awesome discovery! And another one of those "many hours of work, single line of fix". Great effort here!

**Comment by swissspidy** [comment]
An upstream PR and then a ticket here to update it & remove the patch would be nice

**Comment by merapi** [comment]
> An upstream PR and then a ticket here to update it & remove the patch would be nice

Upstream: https://github.com/bubkoo/html-to-image/pull/194
Remove patch: https://github.com/google/web-stories-wp/issues/9229

**Pull request merged**